base/resources/scripts/languagecheck.js

```diff
@@ -0,0 +1,61 @@
+/* Nodejs to set up a server on 42536 to modify the argv.json file
+   used by in-browser VSCode. Uses the accept-language header.
+   Also used by jupyter to determine language.
+ */
+var http = require("http");
+var fs = require("fs");
+
+http
+  .createServer(function (req, res) {
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    const SUPPORTED_LANGS = ["en", "fr"];
+    const DEFAULT_LANG = "en";
+    let lang = (req.headers["accept-language"] || DEFAULT_LANG).substring(0, 2);
+    if (!SUPPORTED_LANGS.includes(lang)) {
+      lang = DEFAULT_LANG;
+    }
+    const browserLang = '"locale":"' + lang + '"'; //used to search
+
+    function readJSONFile(callback) {
+      fs.readFile("/home/jovyan/.config/Code/User/argv.json", "utf-8",
+        function (err, data) {
+          if (err) {
+            return callback(500,"");
+          }
+          callback(200, data);
+        }
+      );
+    }
+
+    function createJSON(callback) {
+      readJSONFile(function detectLangSwitch(returnCode, content) {
+        if (returnCode === 500){
+          return callback(returnCode, "Error detecting language");
+        }
+        if (content.indexOf(browserLang) >= 0) {
+          return callback(200,"No language switch needed");
+        }
+        //Create JSON to use
+        let JSON_DATA = JSON.parse('{ "locale":"en" }');
+        JSON_DATA.locale = lang;
```

Comment:
Almost. The idea here is to `JSON.parse` the actual file content instead of searching for a specific string:
* You can mutate its contents (`locale`) and rewrite it, preserving any other config in the file
* Formatting doesn't matter when matching contents as long as the JSON is valid
```json
{"locale":"en"}
// vs.
{"locale":      "en"}
// vs.
{
  "locale": "en"
}
```